packages/grafana-data/src/types/panel.ts

```diff
@@ -151,3 +153,17 @@ export enum VizOrientation {
   Vertical = 'vertical',
   Horizontal = 'horizontal',
 }
+
+export interface PanelHoverEventPayload {
+  panelId: number;
+  // temp test event
+  pos: {
+    x: number;
+    y: number;
+  };
+}
+
+export class PanelHoverEvent extends BusEventWithPayload<PanelHoverEventPayload> {
+  static readonly type = 'panel-hover-event';
+  public readonly type = PanelHoverEvent.type;
+}
```

Comment:
I wonder if we can't use a similar  approach to this as the actionCreatorFactory?

Reply:
```typescript
export const panelHoverEvent = eventCreatorFactory<BusEventWithPayload<PanelHoverEventPayload>>('panel-hover-event').create()

bus.on(panelHoverEvent, event => {
  events.push(event);
});
bus.emit(panelHoverEvent({panelId: 1, pos: {x:0,y:0}));

```